src/openforms/js/components/admin/form_design/form-creation-form.js

```diff
@@ -685,49 +643,9 @@ function reducer(draft, action) {
       draft.priceRules = updatedRules;
       break;
     }
-    case 'PRICE_RULES_SAVED': {
-      // set the generated UUID from the backend for created rules
-      const createdRules = action.payload;
-      for (const rule of createdRules) {
-        const {uuid, index} = rule;
-        draft.priceRules[index].uuid = uuid;
-      }
-
-      // clear the state of rules to delete, as they have been deleted
-      draft.priceRulesToDelete = [];
-      break;
-    }
 
     /**
-     * Validation error handling
-     */
-    case 'PROCESS_VALIDATION_ERRORS': {
-      const {fieldPrefix, errors} = action.payload;
-      // process the errors with their field names
-      const tabsWithErrors = [];
-      const prefixedErrors = errors.map(err => {
-        const fieldName = err.name.split('.')[0];
-        if (!tabsWithErrors.includes(fieldName))
-          tabsWithErrors.push(FORM_FIELDS_TO_TAB_NAMES[fieldName]);
-        let key;
-        // literals are tracked separately in the state
-        if (fieldPrefix === 'form' && fieldName === 'literals') {
-          key = err.name;
-        } else if (fieldPrefix === 'logic-rules') {
-          key = err.name;
-        } else {
-          key = `${fieldPrefix}.${err.name}`;
-        }
-        return [key, err.reason];
-      });
-      draft.validationErrors = [...draft.validationErrors, ...prefixedErrors];
```

Comment:
I noticed that now if we submit a form and the errors are in the form or form step tab, once the user resolves the errors, the warning icon in the tab disappears.
In the logic tab, this doesn't happen (the icon remains also when all errors are resolved)